vnext/Microsoft.ReactNative/Views/FocusZoneViewManager.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "FocusZoneViewManager.h"
+
+#include "ViewControl.h"
+
+#include "DynamicAutomationProperties.h"
+
+#include <Modules/NativeUIManager.h>
+#include <Utils/AccessibilityUtils.h>
+#include <Utils/Helpers.h>
+#include <Utils/PropertyUtils.h>
+
+#include <INativeUIManager.h>
+#include <IReactInstance.h>
+
+#include <winrt/Windows.System.h>
+#include <winrt/Windows.UI.Core.h>
+
+#include <UI.Xaml.Controls.Primitives.h>
+#include <UI.Xaml.Input.h>
+#include <winrt/Windows.ApplicationModel.Resources.Core.h>
+
+namespace winrt {
+using namespace Windows::Foundation;
+} // namespace winrt
+
+namespace Microsoft {
+namespace ReactNative {
+
+void FocusZoneViewShadowNode::createView() {
+  Super::createView();
+
+  auto panel = GetViewPanel();
+  m_panelKeyDownRevoker =
+      panel.PreviewKeyDown(winrt::auto_revoke, [=](const auto &sender, const auto &e) { OnKeyDown(sender, e); });
```

Comment:
Just for my understanding, why is `PreviewKeyDown` preferred over `KeyDown`? I can't tell the difference from the docs.

Reply:
My initial implementation registered the events inside ScrollViewManager - using the bubbled rather than the routed event resulted in both the XAML element scrolling the content, and our manager moving focus to the next element (e.handled was not getting set early enough). Given that the current implementation doesn't modify the ScrollViewManager but rather does use the FocusZone component, the preview event might no longer be necessary.

Reply:
The Tab behavior is still broken however in this implementation - pressing tab while there are more elements than currently visible in the list acts as if PgDown had been pressed, not moving focus outside the FlatList. I'm working with Keith and others on figuring out how to do this.